akka-actor/src/main/scala/akka/serialization/Serialization.scala

```diff
@@ -134,11 +156,22 @@ class Serialization(val system: ExtendedActorSystem) extends Extension {
   val log: LoggingAdapter = _log
   private val manifestCache = new AtomicReference[Map[String, Option[Class[_]]]](Map.empty[String, Option[Class[_]]])
 
+  /** INTERNAL API */
+  @InternalApi private[akka] def serializationInformation: Serialization.Information =
+    system.provider.serializationInformation
+
   /**
    * Serializes the given AnyRef/java.lang.Object according to the Serialization configuration
    * to either an Array of Bytes or an Exception if one was thrown.
    */
-  def serialize(o: AnyRef): Try[Array[Byte]] = Try(findSerializerFor(o).toBinary(o))
+  def serialize(o: AnyRef): Try[Array[Byte]] = {
+    val oldInfo = Serialization.currentTransportInformation.value
```

Comment:
Might be worth pulling this out to a function e.g. 

```scala
  private def withTransportSet[T](f: () => T): T = {

    val oldInfo = Serialization.currentTransportInformation.value
    try {
      if (oldInfo eq null)
        Serialization.currentTransportInformation.value = serializationInformation
      f()
    } finally {
      Serialization.currentTransportInformation.value = oldInfo
    }
  }
```

Then

```
  def serialize(o: AnyRef): Try[Array[Byte]] = {
    withTransportSet { () =>
      Try(findSerializerFor(o).toBinary(o))
    }
  }
```

Reply:
the drawback is the extra allocation, perhaps I'm too concerned about allocations and doesn't matter for these methods.